Feeders/refactorization for the CLI

albertinisg opened the pull request:
This PR hangs from #116 (which was already LGTM'd)

This PR includes the following changes:
- Split geocoder from restaurant feeder and add geocoding objects in a JSON file (as I mentioned in https://github.com/Fiware/tutorials.TourGuide-App/issues/63#issuecomment-205840078)
- Adapted feeders to work with the CLI interface including new parameters.
- Removed the pre-loaded image. Now all the data should be loaded using the feeders.
- Updated README.md with the feeders documentation.

Commit:
Split restaurantfeeder.js into a single feeder and a geocoder script

Commit:
Update restaurants and geocoder objects files

Commit:
Remove non used function

Commit:
Update utils.js functions for the feeders

Commit:
Refactor feeders for being launched with the CLI

Commit:
Update README.md

Commit:
Remove pre-loaded data image

Now everything should be loaded using the CLI (feed restaurants, reviews and reservations)

Commit:
Update tests

As the mongodb database is now empty, we need to generate a restaurant in both reviews or reservations

FiwareULPGC commented on server/feeders/reservationsgenerator.js:
```diff
@@ -4,111 +4,231 @@
+    showHelp();
+  }
+
+  if (typeof argv.fromDate === 'string' &&
```
Fits in one line.

FiwareULPGC commented on server/feeders/reservationsgenerator.js:
```diff
@@ -4,111 +4,231 @@
+    fromDate = argv.fromDate;
+  }
+
+  if (typeof argv.toDate === 'string' &&
```
Fits in one line.

FiwareULPGC commented on server/feeders/reservationsgenerator.js:
```diff
@@ -4,111 +4,231 @@
+    toDate = argv.toDate;
+  }
+
+  if (typeof argv.restaurant === 'string' &&
```
Fits in one line.

FiwareULPGC commented on server/feeders/restaurantGeocoder.js:
```diff
@@ -0,0 +1,179 @@
+    showHelp();
+  }
+
+  if (typeof argv.jsonfile === 'string' &&
```
Fits in one line.

FiwareULPGC commented on server/feeders/restaurantGeocoder.js:
```diff
@@ -0,0 +1,179 @@
+    cacheFile = argv.jsonfile;
+  }
+
+  if (typeof argv.output === 'string' &&
```
Fits in one line.

FiwareULPGC commented on server/feeders/restaurantGeocoder.js:
```diff
@@ -0,0 +1,179 @@
+*/
+function exportGeocode(restaurants) {
+
+  //restaurantsData = restaurantsData.slice(0,5);
```
Commented code.

FiwareULPGC commented on server/feeders/restaurantfeeder.js:
```diff
@@ -4,106 +4,180 @@
+    showHelp();
+  }
+
+  if (typeof argv.jsonfile === 'string' &&
```
Fits in one line.

FiwareULPGC commented on server/feeders/restaurantfeeder.js:
```diff
@@ -4,106 +4,180 @@
+    cacheFile = argv.jsonfile;
+  }
+
+  if (typeof argv.geofile === 'string' &&
```
Fits in one line.

FiwareULPGC commented on server/feeders/restaurantfeeder.js:
```diff
@@ -4,106 +4,180 @@
+    geolocationFile = argv.geofile;
+  }
+
+  if (typeof argv.restaurant === 'string' &&
```
Fits in one line.

FiwareULPGC commented on server/feeders/restaurantfeeder.js:
```diff
@@ -4,106 +4,180 @@
+ * Function to read the restaurant information from a given file
+*/
+function loadRestaurantData() {
+
```
Unneeded line.

FiwareULPGC commented on server/feeders/restaurantfeeder.js:
```diff
@@ -4,106 +4,180 @@
+ * Function that adapts the information for Orion Context Broker
+*/
+function feedOrionRestaurants() {
+
```
Unneeded line.

FiwareULPGC commented on server/feeders/restaurantfeeder.js:
```diff
@@ -121,21 +195,21 @@ var feedOrionRestaurants = function() {
+  var capacity = [50, 80, 100, 120, 160, 200];
+
+  restaurantsData.forEach(function(element, index) {
 
```
Unneeded line.

FiwareULPGC commented on server/feeders/reviewsgenerator.js:
```diff
@@ -36,133 +38,259 @@ var fiwareHeaders = {
-    }, returnPost);
-  });
-};
+  if (typeof argv.restaurant === 'string' &&
```
Fits in one line.

FiwareULPGC commented on server/feeders/reviewsgenerator.js:
```diff
@@ -36,133 +38,259 @@ var fiwareHeaders = {
     });
-  });
-};
+  }  else if (restaurantSelected) {
```
May be better to put the if inside {} ? It is hard to follow the lines 146-159.

albertinisg replied:
It's already inside brackets :)

FiwareULPGC replied:
We referred that could be clearer to put all the if block inside curly braces (`{` `}`) instead of use an _elif_ like style.

You have:

```
if (organization) {
  ...
}
else if (restaurantSelected) {
  ...
} else {
  ...
}
```

and we suggest:

```
if (organization) {
  ...
}
else {
  if (restaurantSelected) {
    ...
  } else {
    ...
  }
}
```

This is only a code style suggestion, not an error. :)

albertinisg replied:
All right! Didn't get it before, thanks for the clarification. I've updated the latest commit adding the suggested change :)

FiwareULPGC commented on server/idas/ul20.js:
```diff
@@ -286,6 +327,15 @@ function initializeSensor(restaurant, room, type) {
+ * @return {Promise} Response of the IoT Agent.
+*/
 function updateSensor(deviceId, type, value, servicePath) {
   switch (type) {
```
Indentation.

FiwareULPGC commented on server/idas/ul20.js:
```diff
@@ -235,43 +253,66 @@ function sendObservation(deviceId, data, servicePath) {
+*/
 function initializeSensor(restaurant, room, type) {
   var deviceId = getDeviceId(restaurant, room, type);
   switch (type) {
```
Indentation.

FiwareULPGC commented on server/utils.js:
```diff
@@ -195,60 +131,82 @@ function fixedEncodeURIComponent(str) {
+ * @return {Object} sensor - Sensor with the units added
+*/
 function setSchemaUnits(sensor) {
   switch (sensor.additionalType) {
```
Indentation.

FiwareULPGC commented:
We tried to test this PR combining it with the PR with the CLI feature, but we couldn't because the load command is not implemented yet.
<img width="1029" alt="load_not_implemented" src="https://cloud.githubusercontent.com/assets/9944828/17923740/87853886-69dd-11e6-9868-3d667b6ca309.png">

Commit:
Minor indentation and style bugfixes

albertinisg commented:
Changes addressed in 8e7cc37434fd6e6fd637022c8e13b65bd1e8726e

As the CLI feature comes in another PR, I suggest to test that there. To test the new feeders, you can follow the new [README.md](https://github.com/albertinisg/tutorials.TourGuide-App/blob/8e7cc37434fd6e6fd637022c8e13b65bd1e8726e/server/feeders/README.md) to load the data.

Lenijas commented on server/feeders/reviewsgenerator.js:
```diff
@@ -36,133 +38,259 @@ var fiwareHeaders = {
     });
-  });
-};
+  }  else if (restaurantSelected) {
```
We referred that could be clearer to put all the if block inside curly braces (`{` `}`) instead of use an *elif* like style.

You have:

```
if (organization) {
  ...
}
else if (restaurantSelected) {
  ...
} else {
  ...
}
```

and we suggest:

```
if (organization) {
  ...
}
else {
  if (restaurantSelected) {
    ...
  } else {
    ...
  }
}
```

This is only a code style suggestion, not an error. :)

Commit:
Add one restaurant to the file at a time

Commit:
Update geo-location file with updated hashes

Commit:
Use 'Franchise1-4' into the Organization field

FiwareULPGC commented:
Apart of the behavior with the franchises explained in #130, if it looks good to @jmcanterafonseca, this PR LGTM

Commit:
Modify the behavior of /api/orion/reviews/organization/:org

As ULPGC team requested, the behavior of the route was not clear and it was not retrieving what it was expected. Now when doing requests to this endpoint, it will return the list of the reviews generated regarding a Franchise

albertinisg commented:
New changes added in a7aec37f7c0eaa362f846cfe54aff6f0a0306edf fixing the Franchises behavior #136

dmabtrg committed:
Use fixed franchises for the restaurants.

dmabtrg commented:
I've added a function to fix the franchise of the restaurants upon creation.  The franchise is selected based on the restaurant Id.

FiwareULPGC commented:
LGTM

dmabtrg reviewed (approved):
LGTM
